src/pybind/mgr/diskprediction/common/db.py

```diff
@@ -0,0 +1,492 @@
+            self.module.get('df').get('stats', {}).get('total_avail_bytes')
+        total_avail_size = float(total_avail_bytes) / (1024 * 1024 * 1024)
+        return round(total_avail_size, 2)
+
+    def get_global_raw_used_size(self):
+        total_used_bytes = \
+            self.module.get('df').get('stats', {}).get('total_used_bytes')
+        total_raw_used_size = float(total_used_bytes) / (1024 * 1024 * 1024)
+        return round(total_raw_used_size, 2)
+
+    def get_global_raw_used_percent(self):
+        total_bytes = \
+            self.module.get('df').get('stats').get('total_bytes')
+        total_used_bytes = \
+            self.module.get('df').get('stats').get('total_used_bytes')
+        if total_bytes and total_used_bytes:
+            total_used_percent = \
+                float(total_used_bytes) / float(total_bytes) * 100
+        else:
+            total_used_percent = 0.0
+        return round(total_used_percent, 2)
+
+    def get_osd_data(self):
+        return self.module.get('config').get('osd_data', '')
+
+    def get_osd_journal(self):
+        return self.module.get('config').get('osd_journal', '')
+
+    def get_osd_metadata(self, osd_id=None):
+        if osd_id is not None:
+            return self.module.get('osd_metadata')[str(osd_id)]
+        return self.module.get('osd_metadata')
+
+    def get_mgr_metadata(self, mgr_id):
+        return self.module.get_metadata('mgr', mgr_id)
+
+    def get_osd_epoch(self):
+        return self.module.get('osd_map').get('epoch', 0)
+
+    def get_osds(self):
+        return self.module.get('osd_map').get('osds', [])
+
+    def get_max_osd(self):
+        return self.module.get('osd_map').get('max_osd', '')
+
+    def get_osd_pools(self):
+        return self.module.get('osd_map').get('pools', [])
+
+    def get_pool_bytes_used(self, pool_id):
+        bytes_used = None
+        pools = self.module.get('df').get('pools', [])
+        for pool in pools:
+            if pool_id == pool['id']:
+                bytes_used = pool['stats']['bytes_used']
+        return bytes_used
+
+    def get_cluster_id(self):
+        return self.module.get('mon_map').get('fsid')
+
+    def get_health_status(self):
+        health = json.loads(self.module.get('health')['json'])
+        return health.get('status')
+
+    def get_mons(self):
+        return self.module.get('mon_map').get('mons', [])
+
+    def get_mon_status(self):
+        mon_status = json.loads(self.module.get('mon_status')['json'])
+        return mon_status
+
+    def get_osd_smart(self, osd_id, device_id=None):
+        result = CommandResult("")
+        data_jaon = {}
+        if str(osd_id).isdigit():
+            osd_name = "osd.%s" % osd_id
+        else:
+            osd_name = osd_id
+        if osd_name[0:4] != 'osd.':
+            raise Exception('not a valid <osd.NNN> id or number')
+
+        if device_id:
+            self.module.send_command(result, 'osd', str(osd_name[4:]), json.dumps({

```

Comment:
I'm not sure why this is fetching the data directly from the OSD.  The devicehealth module already has a scrape method that scrapes results from all devices.  Can we reuse that data instead?

Reply:
@liewegas Yes, I already reuse devicehealth module data. But if the devicehealth module not enabled, the diskpredection module have secondary option to get smart data from osd daemon. Does the diskprediction module limited to use the devicehealth data?

Reply:
the devicehealth module will always be enabled (an 'always on' ability is coming shortly) so it can be relied on

Reply:
Got it, the diskprediction module change to related on the devicehealth module.